File: musify/src/main/java/ro/ubb/postuniv/musify/dto/SearchViewDto.java
package ro.ubb.postuniv.musify.dto;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SearchViewDto {

    private List<PlaylistViewDto> playlists;
    private List<ArtistViewDto> artists;
    private List<BandDto> bands;
    private List<AlbumViewDto> albums;
    private List<SongViewDto> songs;
}

File: webapp/src/app/models/searchViewDto.ts
import {SongViewDto} from "./songViewDto";
import {AlbumViewDto} from "./albumViewDto";
import {PlaylistViewDto} from "./playlistViewDto";

export interface SearchViewDto {
  playlists: PlaylistViewDto[];
  songs: SongViewDto[];
  albums: AlbumViewDto[];
}
